ui/guest/src/store/subjects.ts

```diff
@@ -29,6 +29,8 @@ export const escape$ =
     ? new Subject<KeyboardEvent>()
     : fromEvent<KeyboardEvent>(document, 'keydown').pipe(filter((e) => e.key === 'Escape'));
 
+export const guestCheckIn$ = new ReplaySubject<boolean>(1);
```

Comment:
There's one detail here (in the usage). We still need to listen for the `xb:loaded` so were sure we have `craftercms.xb`, then subscribe to `guestCheckIn$`